### PR TITLE
Add SAM template

### PR DESCRIPTION
**Pull request by anchen1011**
Here we provide SAM template with API template (POST file, GET url), and there usage.

### CODE REVIEW

File: docs/Overview.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+             /lib/*
+```
+
+## System Design
+
+**Serverless Model Serving**
+
+First models are trained and exported into model zoo, ONNX or saved as checkpoint (CKPT). Then we use Lambda Inference Toolkit to pack the model into a package that's Inference + Lambda enabled (*details on this part explained in the next graph*). Then the deployed package will be served on AWS Lambda with charge by query, reasonable response time, and effortless and limitless scalability. The serverless model serving could be triggered by AWS Services (S3, API Gateway, Kinesis, Dynamo, Kinesis, etc.), API Callbacks, HTTP, Mobile, etc., and the serving itself is also supported by AWS services (together with Java, Python, etc.).
+
+**Lambda Inference Toolkit**
```
Comment (nswamy) [inline docs/Overview.md]: this can be removed based on our earlier discussion and feedback from Sunil

File: sam/template.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      StageName: prod
+      Variables:
+        LambdaFunctionName: !Ref LambdaFunction
+
+  LambdaFunction:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      Handler: lambda_function.lambda_handler
+      Runtime: python2.7
+      CodeUri: ./lambda_function.zip #UPDATE
```
Comment (kalyc) [inline sam/template.yaml]: How about storing the zip file on S3 & adding that here instead?
So, parameters would be:
`Code:
  S3Bucket:
 S3Key:
`
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html

File: docs/SAM-Usage.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+Setup more resources/services according to [SAM Doc](https://github.com/awslabs/serverless-application-model/blob/master/HOWTO.md) or [CloudFormation Doc](https://aws.amazon.com/documentation/cloudformation/)
+
+For example, setup a POST file API Gateway with demo swagger file
+```
+# swagger.yaml
+# <<region>> : AWS region set in Pre-Requisites, referenced twice in swagger.yaml
+# <<accountId>> : your global AWS account ID (found in MyAccount)
+uri: arn:aws:apigateway:<<region>>:lambda:path/2015-03-31/functions/arn:aws:lambda:<<region>>:<<accountId>>:function:${stageVariables.LambdaFunctionName}/invocations
+
+# template.yaml
```
[the remark this answers is not in the document]
Comment (anchen1011) [reply docs/SAM-Usage.md]: `template.yaml` is by [sam convention](https://github.com/awslabs/aws-sam-cli#usage)

File: sam/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+swagger: "2.0"
```
[the remark this answers is not in the document]
Comment (anchen1011) [reply sam/swagger.yaml]: I don't think so. [ref](https://github.com/awslabs/aws-sam-cli/blob/develop/tests/integration/testdata/start_api/swagger-template.yaml)

File: sam/template.yaml (hunk 2 above)
Comment (anchen1011) [reply sam/template.yaml]: SAM actually do that for you with `aws cloudformation package` which automatically upload your `lambda_function.zip` to S3 bucket and generate a `template-out.yaml` which is the actual cloudformation that is used to define the stack. 

Good idea though!

File: docs/Discussion.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+## Sagemaker, SAM and MMS
+
+Sagemaker https://aws.amazon.com/cn/sagemaker/ is a stable AWS official tool to train and deploy (on server) machine learning models.
```
[the remark this answers is not in the document]
Comment (anchen1011) [reply docs/Discussion.md]: Thanks! Documentation will be updated and reorganized in a separate PR, where all the comments will be incorporated!

File: docs/SAM-Quickstart.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+- Get the URL of the Serverless application that was created
+
+```
+aws cloudformation describe-stacks --stack-name <STACK_NAME> | python -c 'import json,sys;obj=json.load(sys.stdin);print obj["Stacks"][0]["Outputs"][0]["OutputValue"];'
+```
+
+- Test with POST request
+
+```
+wget https://s3.us-east-2.amazonaws.com/baiachen-amazon-ai-work-data/cat.png
```
[the remark this answers is not in the document]
Comment (anchen1011) [reply docs/SAM-Quickstart.md]: Thanks! Documentation will be updated and reorganized in a separate PR, where all the comments will be incorporated!

File: docs/Discussion.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+## Sagemaker, SAM and MMS
+
+Sagemaker https://aws.amazon.com/cn/sagemaker/ is a stable AWS official tool to train and deploy (on server) machine learning models. 
+
+SAM-CLI https://github.com/awslabs/aws-sam-cli is a stable AWS Labs official tool that supports local Lambda testing.
```
[the remark this answers is not in the document]
Comment (anchen1011) [reply docs/Discussion.md]: Thanks! Documentation will be updated and reorganized in a separate PR, where all the comments will be incorporated!

File: docs/Discussion.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+## Sagemaker, SAM and MMS
+
+Sagemaker https://aws.amazon.com/cn/sagemaker/ is a stable AWS official tool to train and deploy (on server) machine learning models. 
+
+SAM-CLI https://github.com/awslabs/aws-sam-cli is a stable AWS Labs official tool that supports local Lambda testing. 
+
+MMS https://github.com/awslabs/mxnet-model-server is an AWS Labs official tool that supports server-based model serving. 
+
+All these tools are very amazing, and we are not trying to reinvent the wheel. We support packing, deploying and serving machine learning models in a serverless manner. We don't support training, deploying with server or local Lambda testing.
```
[the remark this answers is not in the document]
Comment (anchen1011) [reply docs/Discussion.md]: Thanks! Documentation will be updated and reorganized in a separate PR, where all the comments will be incorporated!

**Pull request merged**